src/compiler/crystal/syntax/lexer.cr

```diff
@@ -1542,57 +1542,61 @@ module Crystal
       when :i64
         gen_check_int_fits_in_size Int64, to_u64, 19
       when :u64
+        gen_check_uint_fits_in_size UInt64, 19
+      when :i128
+        gen_check_int_fits_in_size Int128, to_u128, 39
+      when :u128
         if negative
-          raise "Invalid negative value #{string_value} for UInt64"
+          raise "Invalid negative value #{string_value} for UInt128"
         end
 
-        check_value_fits_in_uint64 string_value, num_size, start
+        check_value_fits_in_uint128 string_value, num_size, start
       end
     end
 
     def deduce_integer_kind(string_value, num_size, negative, start)
       if negative
-        check_negative_value_fits_in_int64 string_value, num_size, start
+        check_negative_value_fits_in_int128 string_value, num_size, start
       else
-        check_value_fits_in_uint64 string_value, num_size, start
+        check_value_fits_in_uint128 string_value, num_size, start
       end
 
       if num_size >= 10
         int_value = absolute_integer_value(string_value, negative)
 
+        int128max = Int128::MAX.to_u128
+        int128max += 1 if negative
+
         int64max = Int64::MAX.to_u64
         int64max += 1 if negative
 
-        int32max = Int32::MAX.to_u32
-        int32max += 1 if negative
-
-        if int_value > int64max
-          @token.number_kind = :u64
-        elsif int_value > int32max
-          @token.number_kind = :i64
```

Comment:
Hmm, what's the reason for removal?

Reply:
the max is now 128 so the line is here now.
https://github.com/crystal-lang/crystal/pull/7093/files#diff-627df40ff6c553c599dce26c4eeb3fb3R1576

Reply:
IIUC int type should be determined dynamically

Reply:
can you give an example.  This seems like an update of the int max?